datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/health/EntityHealthResolver.java

```diff
@@ -0,0 +1,328 @@
+      @Nonnull final GraphClient graphClient,
+      @Nonnull final TimeseriesAspectService timeseriesAspectService) {
+    this(entityClient, graphClient, timeseriesAspectService, new Config(true, true));
+  }
+
+  public EntityHealthResolver(
+      @Nonnull final EntityClient entityClient,
+      @Nonnull final GraphClient graphClient,
+      @Nonnull final TimeseriesAspectService timeseriesAspectService,
+      @Nonnull final Config config) {
+    _entityClient = entityClient;
+    _graphClient = graphClient;
+    _timeseriesAspectService = timeseriesAspectService;
+    _statusCache =
+        CacheBuilder.newBuilder().maximumSize(0).expireAfterWrite(1, TimeUnit.MINUTES).build();
+    _config = config;
+  }
+
+  @Override
+  public CompletableFuture<List<Health>> get(final DataFetchingEnvironment environment)
+      throws Exception {
+    final Entity parent = environment.getSource();
+    return CompletableFuture.supplyAsync(
+        () -> {
+          try {
+            final CachedHealth cachedStatus =
+                _statusCache.get(
+                    parent.getUrn(),
+                    () -> (computeHealthStatusForAsset(parent.getUrn(), environment.getContext())));
+            return cachedStatus.healths;
+          } catch (Exception e) {
+            throw new RuntimeException("Failed to resolve asset's health status.", e);
+          }
+        });
+  }
+
+  /**
+   * Computes the "resolved health status" for an asset by
+   *
+   * <p>- fetching active (non-deleted) assertions - fetching latest assertion run for each -
+   * checking whether any of the assertions latest runs are failing
+   */
+  private CachedHealth computeHealthStatusForAsset(
+      final String entityUrn, final QueryContext context) {
+    final List<Health> healthStatuses = new ArrayList<>();
+
+    if (_config.getIncidentsEnabled()) {
+      final Health incidentsHealth = computeIncidentsHealthForAsset(entityUrn, context);
+      if (incidentsHealth != null) {
+        healthStatuses.add(incidentsHealth);
+      }
+    }
+
+    if (_config.getAssertionsEnabled()) {
+      final Health assertionsHealth = computeAssertionHealthForAsset(entityUrn, context);
+      if (assertionsHealth != null) {
+        healthStatuses.add(assertionsHealth);
+      }
+    }
+
+    return new CachedHealth(healthStatuses);
+  }
+
+  /**
+   * Returns the resolved "incidents health", which is currently a static function of whether there
+   * are any active incidents open on an asset
+   *
+   * @param entityUrn the asset to compute health for
+   * @param context the query context
+   * @return an instance of {@link Health} for the entity, null if one cannot be computed.
+   */
+  private Health computeIncidentsHealthForAsset(
+      final String entityUrn, final QueryContext context) {
+    try {
+      final Filter filter = buildIncidentsEntityFilter(entityUrn, IncidentState.ACTIVE.toString());
+      final SearchResult searchResult =
+          _entityClient.filter(
```

Comment:
If caching is needed, the systemEntityClient and a slight variation of the call should provide that cache without having to maintain a separate cache.

Reply:
I see, the cache is on the higher level object  including data from those other services. Disregard the comment above.

Reply:
im removing the cache. it's only really required in OSS because it's expensive to compute assertion health. but its not being used anyways so im removing as you've mentioned